RealmModelGenerator/EntitiesViewController.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  EntitieViewController.swift
+//  RealmModelGenerator
+//
+//  Created by Zhaolong Zhong on 3/21/16.
+//  Copyright © 2016 QuarkWorks. All rights reserved.
+//
+
+import Cocoa
+
+
+class EntitiesViewController: NSViewController, EntitiesViewDelegate, EntitiesViewDataSource {
+    static let TAG = NSStringFromClass(EntitiesViewController)
+    
+    @IBOutlet weak var entitiesView: EntitiesView!
+    
+    private var schema = Schema();
+    private var model: Model?
+    private var entities:[Entity] = []
+    
+    var defaultSchema: Schema?{
+        willSet(defaultSchema) {
+            schema = defaultSchema!;
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view.
+        
+        entitiesView.delegate = self
+        entitiesView.dataSource = self
+        
+        if let currentModel = schema.getCurrentModel() {
+            model = currentModel
+        } else {
+            model = schema.createModel()
+        }
+        
+        entities = model!.entities
+    }
+    
+    //MARK: EntitiesViewDelegate
+    func entitiesView(entitiesView:EntitiesView, titleForEntityAtIndex index:Int) -> String? {
+        return entities[index].name
+    }
+    
+    @objc func entitiesView(entitiesView:EntitiesView, didRemoveEntityAtIndex index:Int) {
+        //TODO: notify attribute, relationship, and entity detail data source change
+    }
+    
+    func numberOfRowsInEntitiesView(entitiesView: EntitiesView) -> Int {
+        return entities.count
+        
+    }
+    
+    func addEntityInEntitiesView(entitiesView: EntitiesView) {
+        entities.append(model!.createEntity())
+    }
+    
+    @objc func removeEntityInEntitiesView(entitiesView: EntitiesView, index:Int) {
+        if (index >= 0 && index < entities.count) {
+            model!.removeEntity(entities[index])
+            entities.removeAtIndex(index)
```

Comment:
Arrays and Dictionaries are structs in swift vs classes. Basically what that means is that this viewController should only hold onto the reference of the model and not [entities]